game-core/src/test/java/games/strategy/triplea/delegate/MatchesTest.java

```diff
@@ -254,4 +266,179 @@ void shouldNotMatchWhenSeaTerritoryIsUnownedAndDoesNotHaveTerritoryAttachment()
       assertThat(newMatch(), notMatches(seaTerritory));
     }
   }
+
+  @Nested
+  @ExtendWith(MockitoExtension.class)
+  final class UnitCanBeInBattle {
+
+    @Mock GamePlayer player;
+    GameData gameData;
+
+    @BeforeEach
+    void setupGameData() {
+      gameData = givenGameData().build();
+    }
+
+    @Test
+    void infrastructureShouldNormallyNotBeInBattle() {
+      when(gameData.getDiceSides()).thenReturn(6);
+      final UnitType unitType = new UnitType("infrastructure", gameData);
+      final UnitAttachment unitAttachment =
+          new UnitAttachment("infrastructure", unitType, gameData);
+      unitAttachment.setIsInfrastructure(true);
+      unitType.addAttachment(UNIT_ATTACHMENT_NAME, unitAttachment);
+      final Unit unit = unitType.create(1, player, true).get(0);
+
+      assertThat(
+          "An infrastructure unit normally can not be in battle",
+          Matches.unitCanBeInBattle(true, true, 1, false, List.of()).test(unit),
```

Comment:
A parameter object I think would help a lot here, this parameter list is not very meaningful:
```true, true, 1, false, List.of()```

I wonder a bit as well why the parameter list is not `Matches.unitTypeCanBeInBattleForRound( < UnitType > , <battleRound:Int> )`

Reply:
Agreed.  `Matches.unitCanBeInBattle` is a complicated matcher that is used in a lot of places.  You'll notice there are 4 overloaded versions of it with different number of parameters.  I first thought about trying to isolate just what I want but it changed too many things that I wasn't sure of.